internal/event/event.go

```diff
@@ -34,11 +34,49 @@ type Event struct {
 }
 
 const (
-	TypeState           = "state"
-	TypeAcknowledgement = "acknowledgement"
-	TypeInternal        = "internal"
+	TypeState             = "state"
+	TypeAcknowledgement   = "acknowledgement"
+	TypeCommentAdded      = "commentAdded"
+	TypeCommentRemoved    = "commentRemoved"
+	TypeDowntimeStart     = "downtimeStart"
+	TypeDowntimeEnd       = "downtimeEnd"
+	TypeDowntimeCancelled = "downtimeCancelled"
+	TypeCustom            = "custom"
+	TypeFlappingStart     = "flappingStart"
+	TypeFlappingEnd       = "flappingEnd"
+	TypeInternal          = "internal"
 )
 
+// Validate checks the event. Returns an error if the validation fails
+func (e *Event) Validate() error {
+	if len(e.Tags) == 0 {
+		return fmt.Errorf("invalid event: tags cannot be empty")
+	} else if e.Severity == SeverityNone {
+		if e.Type == "" {
+			return fmt.Errorf("invalid event: must set 'type' or 'severity'")
+		}
+
+		switch e.Type {
+		case TypeState:
+		case TypeAcknowledgement:
+		case TypeCommentAdded:
+		case TypeCommentRemoved:
+		case TypeDowntimeStart:
+		case TypeDowntimeEnd:
+		case TypeDowntimeCancelled:
+		case TypeCustom:
+		case TypeFlappingStart:
+		case TypeFlappingEnd:
+			return nil
```

Comment:
There's no automatic fallthrough between cases, so this doesn't really do what you probably expect (or in some way it does, the next statement after `switch` will be the `return nil` at the end of the function, so you won't notice a difference here.

Hint: you can do `case A, B, C:`